### PR TITLE
Updates from pen test

### PR DESCRIPTION
**Pull request by JimMadge**
Updating documents to reflect the results of the pen test and add past DSGs/production instances.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 1]
```diff
@@ -19,19 +19,24 @@ Refer to the [Deployment](https://data-safe-haven.readthedocs.io/en/latest/deplo
 - [ ] Draft a changelog for the release similar to our previous releases, see https://github.com/alan-turing-institute/data-safe-haven/releases
 - [ ] Deploy an SHM from this branch and save a transcript of the deployment logs
 - [ ] Build an SRE compute image (SRD) and save transcripts of the logs
-- Using the new image, deploy two SREs which between them cover tiers 2 and 3 and Guacamole/Microsoft RDS
+- Using the new image, deploy two SREs which between them cover tiers 2 and 3
```
Comment (jemrobinson) [inline .github/ISSUE_TEMPLATE/release_checklist.md]: Maybe simpler as "Using the new image, deploy a tier 2 and a tier 3 SRE"?

**Review by jemrobinson — APPROVED** [review]
LGTM. One minor comment.

**Comment by JimMadge** [comment]
@alan-turing-institute/data-study-group @dsj976 would appreciate a check of the dates/versions of DSH used in production :pray:.

**Comment by jemrobinson** [comment]
David is on leave at the moment, as is Helen. Possibly @edwardchalstrey1, @edchapman88 or @cptanalatriste can help here?

**Comment by cptanalatriste** [comment]
@JimMadge , on the last DSG we used DSH `v4.1.0`

File: VERSIONING.md [hunk 2]
```diff
@@ -76,6 +90,7 @@ The following versions have been evaluated by third party security consultants p
...
 | [v3.4.0](https://github.com/alan-turing-institute/data-safe-haven/releases/tag/v3.4.0) | 22 April 2022    | Penetration test evaluating (1) external attack surface, (2) ability to exfiltrate data from the system, (3) ability to transfer data between SREs, (4) ability to escalate privileges on the SRD. | No major security issues identified. |
 | [v4.0.0](https://github.com/alan-turing-institute/data-safe-haven/releases/tag/v4.0.0) | 2 September 2022 | Penetration test evaluating ability to infiltrate/exfiltrate data from the system.                                                                                                                 | No major security issues identified. |
+| [v4.2.0](https://github.com/alan-turing-institute/data-safe-haven/releases/tag/v4.2.0) | 22 March 2024    | Penetration test evaluating ability to infiltrate/exfiltrate data from the system. Repeat tests for v4.0.0 vulnerabilities.                                                                                                                | No major security issues identified. |
```
Comment (jemrobinson) [inline VERSIONING.md]: Do you want to add the `v5.0.0-rc.1` pen test in here? It was done in September/October 2023.

**Pull request merged**